Limit AsyncAdaptedQueue to Python 3.7

zzzeek opened the pull request:
Tests here are failing for python 3.6 due to the lack
of asyncio.run().   It seems to be non-trivial to vendor
a working version of this in Python 3.6 as the tests here
are running it in alternate threads.

Change-Id: I9398c9fb2aa87f3228ce2f59277de732091bd541

<!-- Provide a general summary of your proposed changes in the Title field above -->

### Description
<!-- Describe your changes in detail -->

### Checklist
<!-- go over following points. check them with an `x` if they do apply, (they turn into clickable checkboxes once the PR is submitted, so no need to do everything at once)

-->

This pull request is:

- [ ] A documentation / typographical error fix
	- Good to go, no issue or tests are needed
- [ ] A short code fix
	- please include the issue number, and create an issue if none exists, which
	  must include a complete example of the issue.  one line code fixes without an
	  issue and demonstration will not be accepted.
	- Please include: `Fixes: #<issue number>` in the commit message
	- please include tests.   one line code fixes without tests will not be accepted.
- [ ] A new feature implementation
	- please include the issue number, and create an issue if none exists, which must
	  include a complete example of how the feature would look.
	- Please include: `Fixes: #<issue number>` in the commit message
	- please include tests.

**Have a nice day!**

zzzeek committed:
Limit AsyncAdaptedQueue to Python 3.7

Tests here are failing for python 3.6 due to the lack
of asyncio.run().   It seems to be non-trivial to vendor
a working version of this in Python 3.6 as the tests here
are running it in alternate threads.

Change-Id: I9398c9fb2aa87f3228ce2f59277de732091bd541

sqla-tester reviewed (commented):
OK, this is **sqla-tester** setting up my work on behalf of **zzzeek** to try to get revision dc096c9b23885a254fcde83fe38efde65b339b40 of this pull request into gerrit so we can run tests and reviews and stuff

sqla-tester commented:
New Gerrit review created for change dc096c9b23885a254fcde83fe38efde65b339b40: https://gerrit.sqlalchemy.org/c/sqlalchemy/sqlalchemy/+/2505

CaselIT committed:
Also fix asyncio exception imports.

The python documentation imports everything directly from the
asyncio package, and it seems that py < 3.8 does not have the
asyncio.exception module

Change-Id: I4719185f02f3c8d030ff24601e15f81767ba7d5a

sqla-tester reviewed (commented):
OK, this is **sqla-tester** setting up my work on behalf of **CaselIT** to try to get revision 35cc1fa3f6ff962676f571ae30851f4b4d96762a of this pull request into gerrit so we can run tests and reviews and stuff

sqla-tester commented:
Patchset 35cc1fa3f6ff962676f571ae30851f4b4d96762a added to existing Gerrit review https://gerrit.sqlalchemy.org/c/sqlalchemy/sqlalchemy/+/2505

sqla-tester reviewed (commented):
**mike bayer** (zzzeek) wrote:

(1 comment)

View this in Gerrit at https://gerrit.sqlalchemy.org/c/sqlalchemy/sqlalchemy/+/2505

* /COMMIT_MSG (line 12): I put this in a PR hoping that github actions would run it on python 3.6 but it looks like that doesn't happen until we get merged to master.

can we add commit message from https://github.com/sqlalchemy/sqlalchemy/pull/5865/commits/35cc1fa3f6ff962676f571ae30851f4b4d96762a here?

sqla-tester reviewed (commented):
**Federico Caselli** (CaselIT) wrote:

(1 comment)

> Patch Set 2:
> 
> (1 comment)

sure

View this in Gerrit at https://gerrit.sqlalchemy.org/c/sqlalchemy/sqlalchemy/+/2505

* /COMMIT_MSG (line 12): The branch name must start with "workflow_test_" for the full pipeline to work

sqla-tester commented:
**Federico Caselli** (CaselIT) wrote:

do we also need a changelog entry? tecnically this is also a bugfix for py<3.8

View this in Gerrit at https://gerrit.sqlalchemy.org/c/sqlalchemy/sqlalchemy/+/2505

sqla-tester commented:
**Federico Caselli** (CaselIT) wrote:

Code-Review+1 Workflow+1

recheck

View this in Gerrit at https://gerrit.sqlalchemy.org/c/sqlalchemy/sqlalchemy/+/2505

sqla-tester commented:
**mike bayer** (zzzeek) wrote:

> Patch Set 3:
> 
> do we also need a changelog entry? tecnically this is also a bugfix for py<3.8

oh because asyncqueuepool would fail w/ 3.6 / 3.7 entirely?   how come we didnt come across that ?

View this in Gerrit at https://gerrit.sqlalchemy.org/c/sqlalchemy/sqlalchemy/+/2505

sqla-tester commented:
**Federico Caselli** (CaselIT) wrote:

> Patch Set 3:
> 
> > Patch Set 3:
> > 
> > do we also need a changelog entry? tecnically this is also a bugfix for py<3.8
> 
> oh because asyncqueuepool would fail w/ 3.6 / 3.7 entirely?   how come we didnt come across that ?

The problem is only on the asyncio.exception module, the asyncio.queue one works also in 3.6 and 3.7. So the bug has been just introduces.
But since the python docs use them directly from the asyncio module I think we should too. https://docs.python.org/3/library/asyncio-queue.html#asyncio.QueueEmpty

View this in Gerrit at https://gerrit.sqlalchemy.org/c/sqlalchemy/sqlalchemy/+/2505

sqla-tester commented:
**mike bayer** (zzzeek) wrote:

Code-Review+2 Workflow+2

let's just put it in, asyncio is alpha so the bar is low  :)

View this in Gerrit at https://gerrit.sqlalchemy.org/c/sqlalchemy/sqlalchemy/+/2505

sqla-tester commented:
Gerrit review https://gerrit.sqlalchemy.org/c/sqlalchemy/sqlalchemy/+/2505 has been **merged**. Congratulations! :)
